cpp/memilio/epidemiology/state_age_function.h

```diff
@@ -176,6 +185,37 @@ struct StateAgeFunction {
         return m_support_max;
     }
 
+    /**
+     * @brief Computes the mean value of the function using the time step size dt and some tolerance tol. 
+     * 
+     * This is a basic version to determine the mean value of a survival function
+     * through numerical integration of the integral that describes the expected value.
+     *
+     * For some specific derivations of StateAgeFunction%s there are more efficient ways to determine the 
+     * the mean value which is why this member function is virtual and can be overridden (see, e.g., ExponentialDecay).
+     * The mean value is only needed for StateAgeFunction%s that are used as TransitionDistribution%s. 
+     *
+     * @param[in] dt Time step size used for the numerical integration. 
+     * @param[in] tol The maximum support used for numerical integration is calculated using this tolerance. 
+     * @return ScalarType mean value.
+     */
+    virtual ScalarType get_mean(ScalarType dt = 1., ScalarType tol = 1e-10)
+    {
+        if (!floating_point_equal(m_mean_tol, tol, 1e-14) || floating_point_equal(m_mean, -1., 1e-14)) {
+            ScalarType mean         = 0.;
+            ScalarType supp_max_idx = std::ceil(get_support_max(dt, tol) / dt);
+            // Integration using Trapezoidal rule.
+            for (int i = 0; i < supp_max_idx; i++) {
+                mean += dt * eval(i * dt);
+            }
+            mean -= 0.5 * dt * eval(0 * dt);
```

Comment:
Why not like this:
`ScalarType mean = 0.5 * dt * eval(0 * dt);`
and then start the for loop at i=1? For me, this would be more intuitive.